eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Pool.hs

```diff
@@ -250,24 +247,15 @@ poolDelegationTransition ::
   ) =>
   TransitionRule (ledger era)
 poolDelegationTransition = do
-  TRC (penv, ps, c) <- judgmentContext
-  case c of
+  TRC (penv, ps, txCert) <- judgmentContext
+  case txCert of
     RegPoolTxCert poolParams -> do
       -- note that pattern match is used instead of cwitness, as in the spec
       poolCertTransition penv ps (RegPool poolParams)
     RetirePoolTxCert sPoolId epochN -> do
       -- note that pattern match is used instead of cwitness, as in the spec
       poolCertTransition penv ps (RetirePool sPoolId epochN)
-    _ | isRegKey c || isDeRegKey c || isDelegation c -> do
-      failBecause $ WrongCertificateTypePOOL 0
-      pure ps
-    _ | isInstantaneousRewards c -> do
-      failBecause $ WrongCertificateTypePOOL 1
-      pure ps
-    _ | isGenesisDelegation c -> do
-      failBecause $ WrongCertificateTypePOOL 2
-      pure ps
```

Comment:
So actually all these 3 cases  (0, 1, 2) are in fact impossible, not just the one below with the comment `impossible case` ? 
Or is it the case that the other changes have rendered them impossible?

Reply:
Yes indeed an impossible case. In the matter of fact, thank you for asking this question, because it is possible to completely remove that case. Which is done in this follow up commit 4ea8e17af3a522d3bf693c631b7a1b74b196cb9f

Reply:
Awesome simplification!  And  the `shrink` is back in `Arbitrary (ShelleyPoolPredFailure era)`